test/integration/file_artifact_test.js

```diff
@@ -228,4 +231,67 @@ suite('artifact extration tests', () => {
     // Missing artifact should have an error...
     assert.equal(result.artifacts['public/my-missing.txt'].storageType, 'error');
   });
+
+  test('upload retry', async () => {
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    worker.on('task run', function() {
+      iptables.reject({
```

Comment:
So I have debated about storing temp files form the container and just initiating another copy operation.  The downside to storing the temp files is now we have a diskspace overhead of storing those until they're fully uploaded, and also making sure we properly remove them.  However it does allow us to do just one copy to not only calculate the hashes of the files (for headers used later on once implemented), but then retrying upload doesn't have to initiate another copy.